zabbix_metrics_watcher.py

```diff
@@ -214,14 +234,32 @@ def run(zabbix_host, osbs_master, config, instance):
 
             logger.info("Found build %s in %s", build_name, status)
             build = Build(build_name, cmd_base)
-            if status == 'Pending':
+            if status == 'New':
+                now = datetime.datetime.now()
+                builds_in_new.setdefault(build_name, now)
+
+            elif status != 'New':
+                try:
+                    del builds_in_new[build_name]
+
+                    # We should reset zabbix item only when the last build in New
+                    # has changed its state
+                    if not builds_in_new:
+                        _send_zabbix_message(zabbix_host, osbs_master, "new_duration", 0)
+                except KeyError:
+                    pass
+
+            elif status == 'Pending':
                 pending.add(build_name)
+
             elif status == 'Running' and changeset in ['added', 'modified']:
                 if build_name in pending:
                     pending_duration = int((build.started_time - build.created_time).total_seconds())
                     _send_zabbix_message(zabbix_host, osbs_master, "pending", pending_duration)
                     logger.info("Pending duration: %s", pending_duration)
                     running_builds.add(build_name)
+                pending.discard(build_name)
```

Comment:
That's possible, although very unlikely - the build usually shows both 'added' and 'modified' rapidly. Gonna fix that